src/ring_theory/localization.lean

```diff
@@ -662,7 +662,9 @@ open is_localization
 
 variables {M}
 
-instance : has_add (localization M) :=
+section
+
+@[irreducible] instance : has_add (localization M) :=
```

Comment:
Sure, I'll do that. But why is this a bad idea specifically?

Reply:
It's a bad idea because I think typeclass search is unable to unify `localization.has_add.add = localization.add_monoid.to_has_add.add` as doing so would require it to reduce `localization.has_add`

Reply:
I tried moving them into irreducible defs, but now I get an error in `localization.lean`. Changing `monoid_of` so it didn't unfold the irreducible defs didn't help either.

Reply:
Finally, making even more things irreducible works!